metricbeat/mb/testing/testdata.go

```diff
@@ -150,14 +151,29 @@ func TestDataFiles(t *testing.T, module, metricSet string) {
 }
 
 // TestDataFilesWithConfig run tests for a testdata config
-func TestDataFilesWithConfig(t *testing.T, module, metricSet string, config DataConfig) {
+func TestDataFilesWithConfig(t *testing.T, module, metricSet string, config DataConfig, testPrefix ...string) {
 	t.Helper()
-	ff, err := filepath.Glob(filepath.Join(config.Path, "*."+config.Suffix))
+
+	// the location of the read files
+	location := filepath.Join(config.Path, "*."+config.Suffix)
+
+	// if this function was called from data_test.go then the testPrefix should be 1
```

Comment:
```suggestion
	// if this function was called from data_test.go then the length of testPrefix should greater than 0.
```
Maybe you mean sth like this?

Reply:
It is also correct, but it will always be one:

```
func getModulesPath() string {
	return "../../../module"
}
```

Why not be specific?

Reply:
I don't think I get it 😞 Could you share what is the actual value of the `testPrefix`, or an example of what it would be?

Reply:
Of course. In the `data_test.go` file we have two functions: `TestAll` and `getModulesPath`. For this specific parameter they are like this:
```
func TestAll(t *testing.T) {
   ...
      mbtest.TestDataFilesWithConfig(t, moduleName, metricSetName, config, getModulesPath())
   ...
}

func getModulesPath() string {
	return "../../../module"
}
```

And we can see that the `TestDataFilesWithConfig` function expects this parameters:
```
func TestDataFilesWithConfig(t *testing.T, module, metricSet string, config DataConfig, testPrefix ...string)
```

From this, we know that when this function is called from `data_test.go`:
```
mbtest.TestDataFilesWithConfig(t, moduleName, metricSetName, config, "../../../module")
```

The string has length 1: ["../../../module"].

If we know look into how this function is called from the modules (example from `collector_test.go`):
```
func TestSameLabels(t *testing.T) {
	dataConfig := mbtest.ReadDataConfig(t, "_meta/samelabeltestdata/config.yml")
	mbtest.TestDataFilesWithConfig(t, "openmetrics", "collector", dataConfig)
}
```
We can see that we only have 4 parameters. So our array will have length 0 (this happens with all modules).

Reply:
I see why it was confusing, you pass the variable as an array with `testPrefix ...string`, but why not to pass it as `string` like `testPrefix string`? It's less confusing it seems :)

Reply:
It is not possible to pass a string as optional, that is why I put it as an array.

Reply:
Hmm, then maybe you cannot make it optional and pass a `""` an empty string in the calls where it is actually empty? This is less confusing when following the code flow I think.

Reply:
That makes it necessary to change every call in each module (it is used). Is this the best way?

Reply:
Ok, I made it mandatory, I am trusting your criteria :)